libdnf-cli/utils/utf8.cpp

```diff
@@ -27,7 +27,7 @@ std::size_t length(const std::string & str) {
     wchar_t wide_char = 0;
 
     while (ptr <= end) {
-        auto bytes = std::mbrtowc(&wide_char, ptr, MB_CUR_MAX, &state);
+        auto bytes = static_cast<int64_t>(std::mbrtowc(&wide_char, ptr, MB_CUR_MAX, &state));
```

Comment:
If you really want to use MB_CUR_MAX, can you change the code like this?:
```
// Maximum length of a multibyte character in the current locale
const auto cur_max = MB_CUR_MAX;

while (ptr <= end) {
    auto bytes = static_cast<int64_t>(std::mbrtowc(&wide_char, ptr, cur_max, &state));
```
Macro MB_CUR_MAX is calling function to determine maximum length of a multibyte character in the current locale. It is waste of time to call it for each character. Use it once before cycle. Please fix it in all functions.

The question is if we need to call the function at all. It is not used in examples, e.g. https://en.cppreference.com/w/cpp/string/multibyte/mbrtowc

Reply:
The proposed solution makes sense.
Using end-ptr is also an option, but I prefer using MB_CUR_MAX, because it seems to be designed for this purposed.